measurements/geography_based.py

```diff
@@ -60,33 +64,69 @@ def geo_ctr(dataloader, args):
 
     pickle.dump(counts, open("results/{}/geo_ctr.pkl".format(args.folder), "wb"))
 
+# private function called from geo_counter() if dataset is of region label form
+def geo_ctr_region(dataloader, args):
+    region_to_id_map = {}
+    id_to_region_map = {}
+
+    for i, (data, target) in enumerate(tqdm(dataloader)):
+        region = target[2][0]
+        id_to_region_map[target[3]] = region
+        if region not in region_to_id_map.keys():
+            region_to_id_map[region] = 0
+        region_to_id_map[region] += 1
+
+    combined_dict = {}
+    combined_dict["region_to_id"] = region_to_id_map
+    combined_dict["id_to_region"] = id_to_region_map
+
+    pickle.dump(combined_dict, open("results/{}/geo_ctr_region.pkl".format(args.folder), "wb"))
```

Comment:
we have a very specific naming convention of geo_{3 letter acronym}, how does geo_ctr_region fit into that? why can't it just be geo_ctr?

Reply:
geo_ctr is still being used, this is just a private function called by geo_ctr when it detects gps data. I thought this would help keep things organized, so geo_ctr doesn't get bloated with the different options. can put this within geo_ctr though if that would make things clearer

Reply:
my problem is with the name of the pickle result file, not the function itself